services/agents/azure_database.go

```diff
@@ -0,0 +1,131 @@
+// pmm-managed
+// Copyright (C) 2017 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+package agents
+
+import (
+	"bytes"
+	"encoding/json"
+	"text/template"
+
+	"github.com/AlekSi/pointer"
+	"github.com/percona/pmm/api/agentpb"
+	"github.com/percona/pmm/api/inventorypb"
+
+	"github.com/percona/pmm-managed/models"
+)
+
+const azureDatabaseTemplate = `---
+active_directory_authority_url: "https://login.microsoftonline.com/"
+resource_manager_url: "https://management.azure.com/"
+credentials:
+  client_id: "{{ .AzureDatabaseClientID}}"
+  client_secret: "{{ .AzureDatabaseClientSecret}}"
+  tenant_id: "{{ .AzureDatabaseTenantID}}"
+  subscription_id: "{{ .AzureDatabaseSubscriptionID}}"
+
+targets:
+resource_groups:
+  - resource_group: "pmmdemo"
+    aggregations:
+      - Average
+{{ .AzureDatabaseResourceTypes }}
+    metrics:
+      - name: "cpu_percent"
+      - name: "memory_percent"
+      - name: "storage_percent"
+      - name: "storage_used"
+      - name: "storage_limit"
+      - name: "network_bytes_egress"
+      - name: "network_bytes_ingress"
+`
+
+// azureDatabaseInstance represents credentials informations.
+type azureDatabaseCredentials struct {
```

Comment:
if we will do what I mentioned above this struct will not be required

Reply:
Looks like we should include ResourceType into AzureOptions.

Reply:
This will duplicate service_type in service table - so we not really need it in AzureOptions

Reply:
Removed this struct.